lib/preprocess/ConfigPreprocessorOverride.ts

```diff
@@ -43,17 +43,24 @@ export class ConfigPreprocessorOverride implements IConfigPreprocessor<Record<st
    * @param handleResponse - Override values that were found for this resource.
    */
   public transform(config: Resource, handleResponse: Record<string, Resource>): IConfigPreprocessorTransform {
-    for (const id of Object.keys(config.properties)) {
-      const overrideValue = handleResponse[id];
-      if (overrideValue) {
-        config.properties[id] = [ overrideValue ];
+    // We know this has exactly 1 result due to the canHandle call
+    const configType = uniqueTypes(config, this.componentResources)[0];
+    const overrideType = handleResponse[IRIS_RDF.type]?.value;
+    // In case the type changes we have to delete all the original properties as those correspond to the old type
```

Comment:
We'll definitely have to document this behaviour.
Because if users define a type that has overlapping properties, they _could_ expect those props to be inherited.
It's ok that this doesn't happen, but we're going to have to be clear upfront about it.

Reply:
> We'll definitely have to document this behaviour.

I'll look into updating the documentation regarding overrides soon-ish

Reply:
Great, thanks :)